zmq/utils/monitor.py

```diff
@@ -70,7 +107,15 @@ def recv_monitor_message(socket: zmq.Socket, flags: int = 0) -> _MonitorMessage:
     -------
     event : dict
         event description as dict with the keys `event`, `value`, and `endpoint`.
+
+    if socket.context is of zmq.asyncio.Context type, return asyncio.Future
     """
+
+    # transparently handle asyncio socket,
+    # returns a future instead of a dict
+    if isinstance(socket.context, zmq.asyncio.Context):
```

Comment:
I don't think this is the way to check for an async socket. Instead, call `msg = socket.recv_multipart` and then check if `msg` is awaitable. If so, wrap the future.

Reply:
i don't know to do it,  please help

Reply:
do you mean like to use 
https://docs.python.org/3/library/inspect.html#inspect.isawaitable
if it is awaitable then , to call the other async function and pass the awaitable variable as an argument, and await it there?

Reply:
like this?
https://github.com/shimondoodkin/pyzmq/blob/asyncmonitorwrap/zmq/utils/monitor.py

Reply:
Yes, that looks right!